File: web-login-core/src/main/java/com/example/weblogincore/domain/model/form/questions/RangeQuestion.java
package com.example.weblogincore.domain.model.form.questions;

public class RangeQuestion extends Question{

    private final double startValue;
    private final double endValue;
    private final double initValue;
    private final Question nextQuestion;

    public RangeQuestion(Long id, String title, String description, String imageUrl, Boolean isMandatory, double startValue, double endValue, double initValue, Question nextQuestion) {
        super(id, title, description, imageUrl, isMandatory);
        this.startValue = startValue;
        this.endValue = endValue;
        this.initValue = initValue;
        this.nextQuestion = nextQuestion;
    }

    @Override
    public Integer totalQuestions() {
        if (nextQuestion == null ) return 1;

        return 1 + nextQuestion.totalQuestions();
    }

    public double getStartValue() {
        return startValue;
    }

    public double getEndValue() {
        return endValue;
    }

    public double getInitValue() {
        return initValue;
    }

    public Question getNextQuestion() {
        return nextQuestion;
    }
}

File: web_login_flutter/lib/models/form/range_question.dart
import 'package:web_login_flutter/models/form/question.dart';

class RangeQuestion extends Question {
  final Question? _nextQuestion;
  final double _initValue;
  final double _startValue;
  final double _endValue;
  const RangeQuestion({required super.id, required super.title, required super.description, required super.imageUrl, required super.isMandatory, nextQuestion, required initValue, required startValue, required endValue}):
        _nextQuestion = nextQuestion, _initValue = initValue, _startValue = startValue, _endValue = endValue;

  Question? get nextQuestion => _nextQuestion;

  bool get hasNextQuestion => _nextQuestion != null;

  @override
  List<Object?> get props => [];

  double get initValue => _initValue;

  double get endValue => _endValue;

  double get startValue => _startValue;

  @override
  String toString() {
    return 'RangeQuestion{_nextQuestion: $_nextQuestion, _initValue: $_initValue, _startValue: $_startValue, _endValue: $_endValue}';
  }
}